src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceSettings.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations;
+
+internal class ResourceSettings : Settings
+{
+    /// <summary>
+    /// Gets or sets the list of enabled resource detectors.
+    /// </summary>
+    public IReadOnlyList<ResourceDetector> EnabledDetectors { get; set; } = [];
+
+    /// <summary>
+    /// Gets or sets the list of enabled resources.
+    /// </summary>
+    public IReadOnlyList<KeyValuePair<string, object>> Resources { get; set; } = [];
```

Comment:
for the future, may want to use union ref struct not to cause object boxing for simple types.